client/jetpack-cloud/controller.ts

```diff
@@ -206,7 +207,11 @@ export async function cloudSiteSelection(
 	}
 
 	if ( siteFragment ) {
-		if ( context.path.startsWith( '/pricing' ) ) {
+		const languages = getLanguageSlugs().join( '|' );
+		// Regex defines url starts with /:locale/pricing or /pricing
+		const pricingMatchingPath = new RegExp( `(^/(${ languages })/pricing)|^/pricing` );
```

Comment:
The regexp could be more compact, like `^(/(${languages}))?/pricing`.

Reply:
Yes, I was going to suggest the same, then thought that it might be more readable 😄